### PR TITLE
The detective's sunglasses turn the world greyscale

### PR DESCRIPTION
**Pull request by MrPerson**
Det spawns with a pair of sunglasses to keep things neater. No new sprites because fuck it.

Picture I guess, it's a gif.
http://www.singulo.io/tg/res/55990.html

### CODE REVIEW

**Comment by Xhuis** [comment]
neat

File: code/game/jobs/job/security.dm [hunk 1]
```diff
@@ -138,6 +138,7 @@ Detective
 	suit = /obj/item/clothing/suit/det_suit
 	gloves = /obj/item/clothing/gloves/color/black
 	head = /obj/item/clothing/head/det_hat
+	glasses = /obj/item/clothing/glasses/sunglasses
```
Comment (Iamgoofball) [inline code/game/jobs/job/security.dm]: forgot to make them noirs fgt

**Comment by as334** [comment]
:+1:

File: code/game/jobs/job/security.dm (hunk 1 above)
Comment (MrPerson) [inline code/game/jobs/job/security.dm]: Noirs are on the map. It's an optional thing

**Comment by Supermichael777** [comment]
holy shit no.
i like color vision thank you.

**Comment by Incoming5643** [comment]
Yeah leaving a pair of non-noirs somewhere nearby is probably a good idea.

That or making noir mode a toggle of the glasses.

**Comment by MrPerson** [comment]
The glasses the det spawns with are normal. If they want noir mode, there's a pair lying in their office. It's completely optional because I know some people won't want it.

**Comment by Jordie0608** [comment]
Should point out this uses a v509 feature hence the compile fail.

**Comment by Jalleo** [comment]
Yay new byond features.

**Comment by Steelpoint** [comment]
I like it, very cool. :+1:

**Comment by Menshin** [comment]
Will the ultra-rare retro noir glasses allow to see NT subliminal messages ("OBEY", "CONSUME" and such) and changeling true form ?

**Comment by RemieRichards** [comment]
@MrPerson you'd have to set footprint ```appearence_flags``` to ignore client colour ```NO_CLIENT_COLOUR``` (or with my PR, ```APPEARANCE_IGNORE_CLIENT_COLOUR```) 
However that'd cause issues with other types of client colouring.

**Comment by JJRcop** [comment]
Overlay in wearer's screen instead of adding the flag to the footprints themselves.

**Comment by RemieRichards** [comment]
@JJRcop can't be done nicely.
that'd require number_of_footprints*8 images, instead of the current 24.
and that's a no.

**Comment by Nienhaus** [comment]
I have been wanting these for 2 years!

**Comment by Supermichael777** [comment]
Actually the glasses highlighting blood and evidence should be a feature. The whole world is panted in black, white, and red.

**Comment by MrPerson** [comment]
@RemieRichards 
Couldn't you just make an exact copy of the footprints via var/image/image = image(footprint); image.appearance = footprint.appearance; footprint.appearance_flags = NO_CLIENT_COLOR?

Then just whenever the blood changes appearance you'd have to update the image(s) as well. And of course add a new HUD for them.

**Comment by RemieRichards** [comment]
yes but that's messy as butts. that's still number_of_footprint images, you've cut it down by a factor of 8, but there's still usually a LOAD of footprints.

plus updating all of those on people's clients? messy.

**Comment by CosmicScientist** [comment]
Christ that looks amazing. Almost wish it could let anything red remain unchanged, not just blood but for anything related to security.

**Comment by MrPerson** [comment]
I def agree that having that many images isn't good, yeah.

I might be able to just let some red through for everything. No promises but I'll try.

In any event I just made this as a quick demo. The code isn't that good and I know it.

**Comment by RemieRichards** [comment]
@MrPerson yeah you can actually just not alter the red component of the red row (first section of the list) and that should let most red leak through.

**Comment by bustygoku** [comment]
Blood should remain red. So should footprints. Anything with fingerprints on it should remain colored buy hey that's just the powergamer in me wanting to take the scanspam out of the det.

**Comment by Jalleo** [comment]
@bustygoku that sounds really laggy.

**Comment by MrPerson** [comment]
The more I look at this code the more I realize how bad it is even as a demo

**Comment by bustygoku** [comment]
Eh this wasn't too bad

**Pull request closed without merging**